x-pack/libbeat/management/manager.go

```diff
@@ -96,24 +95,25 @@ func (cm *Manager) Enabled() bool {
 	return cm.config.Enabled
 }
 
-// Start the config manager
-func (cm *Manager) Start(stopFunc func()) {
-	if !cm.Enabled() {
-		return
-	}
-
+// Start the config manager.
+func (cm *Manager) Start() error {
 	cm.lock.Lock()
 	defer cm.lock.Unlock()
 
+	if !cm.Enabled() {
+		return nil
+	}
+
 	cfgwarn.Beta("Fleet management is enabled")
 	cm.logger.Info("Starting fleet management service")
 
-	cm.stopFunc = stopFunc
 	cm.isRunning = true
 	err := cm.client.Start(context.Background())
 	if err != nil {
 		cm.logger.Errorf("failed to start elastic-agent-client: %s", err)
 	}
+	cm.logger.Info("Ready to receive configuration")
+	return err
```

Comment:
I believe you missed the return when handling the error. 
As I'm suggesting already, I'm always for wrapping errors and giving more context. But it's up to you, just the return on error is the blocker here.

```suggestion
	if err != nil {
		fmt.Errorf("failed to start elastic-agent-client: %w", err)
		cm.logger.Errorf(err.Error())
		return err
	}
	cm.logger.Info("Ready to receive configuration")
	return nil
```

Reply:
I've changed the code not exactly likes your suggestion but I do return the error.